### PR TITLE
Add partial 2sq development in iset.mm

### PR DESCRIPTION
**Pull request by jkingdon**
The lemmas for `2sq` intuitionize with few or no changes until we get to `2sqlem11` which depends on things we don't have yet. Although #4387 if merged would partially close that gap, it doesn't have `lgsqr` and `m1lgs` which are used by the `2sqlem11` proof.

Given that, it seems better to merge what I have now rather than keep it sitting on a long-lived branch where no one else can easily see it.

### CODE REVIEW

**Review by avekens — APPROVED** [review]

**Review by icecream17 — APPROVED** [review]

**Comment by jkingdon** [comment]
The force-push just now was to rebase and resolve merge conflicts.

I think I'll merge this once tests pass to avoid further merge conflicts including perhaps with #4391 .

**Pull request merged**
